client/src/components/Auth/SignIn.jsx

```diff
@@ -56,137 +54,151 @@ width: 70%;
+        Sign In
+      </Heading>
+      <SignInForm />
+      <PasswordForgetLink />
+      <SignUpLink />
     </Box>
+  </Box>
 );
 
 const INITIAL_STATE = {
-    email: '',
-    password: '',
-    error: null,
+  email: "",
+  password: "",
+  error: null
 };
 
 class SignInFormBase extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = { ...INITIAL_STATE };
-    }
-
-    onSubmit = event => {
-        const { email, password } = this.state;
-
-        this.props.firebase
-            .doSignInWithEmailAndPassword(email, password)
-            .then(() => {
-                this.setState({ ...INITIAL_STATE });
-                this.props.history.push(ROUTES.HOME);
-            })
-            .catch(error => {
-                this.setState({ error });
-            });
-
-        event.preventDefault();
-    };
-
-    onChange = event => {
-        this.setState({ [event.target.name]: event.target.value });
-    };
-
-    render() {
-        const { email, password, error } = this.state;
-
-        const isInvalid = password === '' || email === '';
-
-        return (
-            <FormContainer>
-                <StyledHeader>
-                    <i class="fas fa-wifi fa-2x" style={{ color: "gold", marginRight: "14px" }}></i>
-                    <h1 >HiveStack</h1>
-                    <StyledSvg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 200 100" preserveAspectRatio="none">
-                        <circle fill="white" cx="0" cy="100" r="100" />
-                        <circle fill="white" cx="200" cy="100" r="100" />
-                    </StyledSvg>
-                </StyledHeader>
-                <StyledForm onSubmit={this.onSubmit}>
-                    <StyledInput
-                        name="email"
-                        value={email}
-                        onChange={this.onChange}
-                        type="text"
-                        placeholder="Email Address"
-                    />
-                    <StyledInput
-                        name="password"
-                        value={password}
-                        onChange={this.onChange}
-                        type="password"
-                        placeholder="Password"
-                    />
-
-
-                    {error && <p>{error.message}</p>}
-                </StyledForm>
-                <LoginButton disabled={isInvalid} onClick={this.onSubmit} primary label="Sign In">Login</LoginButton>
-            </FormContainer>
-        );
-    }
+  constructor(props) {
+    super(props);
+
+    this.state = { ...INITIAL_STATE };
+  }
+
+  onSubmit = event => {
+    const { email, password } = this.state;
+
+    this.props.firebase
+      .doSignInWithEmailAndPassword(email, password)
+      .then(() => {
+        this.setState({ ...INITIAL_STATE });
+        this.props.history.push(ROUTES.HOME);
+      })
+      .catch(error => {
+        this.setState({ error });
+      });
+
+    event.preventDefault();
```

Comment:
I would suggest to just put the preventDefault at the top to avoid any issues. It should not interfere with your history.push in your promise or the catch rejection.

Reply:
Yes sir, will put it at the top on next PR. Haven't really touched this file so I'll have it changed on the upcoming PR.

Reply:
Completed.